model/task_queue_service_test.go

```diff
@@ -1189,6 +1189,76 @@ func (s *taskDAGDispatchServiceSuite) TestNextTaskForDefaultTaskSpec() {
 	// .....
 }
 
+func (s *taskDAGDispatchServiceSuite) TestIsRefreshFindNextTaskThreadSafe() {
+	s.Require().NoError(db.ClearCollections(task.Collection, distro.Collection, TaskQueuesCollection))
+	d := distro.Distro{
+		Id: "distro_1",
+		DispatcherSettings: distro.DispatcherSettings{
+			Version: evergreen.DispatcherVersionRevisedWithDependencies,
+		},
+	}
+	s.Require().NoError(d.Insert(s.ctx))
+
+	items := []TaskQueueItem{}
+	for i := 0; i < 50; i++ {
+		items = append(items, TaskQueueItem{
+			Id:            fmt.Sprintf("%d", i),
+			BuildVariant:  "variant_1",
+			Version:       "version_1",
+			Project:       "project_1",
+			GroupMaxHosts: 0,
+		})
+		t := task.Task{
+			Id:                fmt.Sprintf("%d", i),
+			BuildVariant:      "variant_1",
+			Version:           "version_1",
+			Project:           "project_1",
+			DistroId:          "distro_1",
+			TaskGroupMaxHosts: 0,
+		}
+		s.Require().NoError(t.Insert())
+	}
+
+	s.taskQueue.Queue = items
+	s.Require().NoError(s.taskQueue.Save())
+	service, err := newDistroTaskDAGDispatchService(s.taskQueue, time.Nanosecond)
```

Comment:
This is to force a rebuild of the queue for every next task operation. This is why this test isn't making an assertion about the number of dispatched tasks like the other thread safety tests in this file.